src/Adaptive.Agrona/Concurrent/UnsafeBuffer.cs

```diff
@@ -257,13 +258,7 @@ public void SetMemory(int index, int length, byte value)
 #endif
         {
             BoundsCheck0(index, length);
-
-            // TODO PERF Naive implementation, we should not write byte by byte, this is slow
-            //UNSAFE.SetMemory(byteArray, addressOffset + index, length, value);
-            for (var i = index; i < index + length; i++)
-            {
-                _pBuffer[i] = value;
-            }
+            Unsafe.InitBlock(((byte*)_pBuffer) + index, value, (uint)length);
```

Comment:
Looking at the IL code:
```
.method public hidebysig static void  InitBlock(void* dst,
                                                  uint8 initValue,
                                                  uint32 size) cil managed aggressiveinlining
  {
    .maxstack  3
    ldarg.0
    ldarg.1
    ldarg.2
    initblk
    ret
  } // end of method Unsafe::InitBlock
```

It doesn't look like it. I wasn't aware of the initblk assumption, so either we make sure the buffer is aligned or we need to modify the IL.